codegen/build.gradle.kts

```diff
@@ -19,3 +19,17 @@ allprojects {
         mavenCentral()
     }
 }
+
+ext {
+    this.set("smithyVersion", project.file("SMITHY_VERSION").readText().replace(System.lineSeparator(), ""))
```

Comment:
why this change? is this some documented smithy convention to have an extra `SMITHY_VERSION` file that then gets read into `gradle.properties`?

Reply:
During the integration testing, it updates the SDK's smithy version to the latest. I could have used `sed -i "25s/.*/smithyVersion=$smithyVersion/" gradle.properties` but using line number is too fragile as dependencies get updated in `gradle.properties` Hence why the extra file seems appropriate.

Reply:
hm im not sure im following, are you saying that with your proposed change, there will be integration testing code that updates the extra file `SMITHY_VERSION`. and if you didnt have this extra file then you would have to update `gradle.properties` via line number using `sed` because in whatever context youre in you cannot access `gradle.properties` via its normal getter/setter gradle api? 

if that understanding is correct, i dont see any code in this PR that writes to a `SMITHY_VERSION` file.